rasa/core/evaluation/stats.py

```diff
@@ -0,0 +1,110 @@
+import os
+from typing import Dict, Text, Union, Any, List, Tuple, TypedDict
+from pathlib import Path
+import json
+import numpy as np
+
+
+class Stats(TypedDict):
+    """A TypedDict for stats."""
+
+    n: int
+    mean: float
+    median: float
+    min: int
+    max: int
+
+
+def load_extracted_markers_json_file(path: Union[Text, Path]) -> List:
+    """Reads a json marker file.
+
+    Args:
+        path: path to a json file.
+    """
+    path = os.path.abspath(path)
+    with open(path) as json_file:
+        extracted_markers = json.load(json_file)
+        return extracted_markers
+
+
+def compute_summary_stats(data_points: Union[List, np.ndarray]) -> Stats:
+    """Computes summary statistics for a given array.
+
+    Computes size, mean, median, min, and max.
+    If size is == 0 returns np.nan for mean, median.
+
+    Args:
+        data_points: can be a numpy array or a list of numbers.
+    """
+    if np.size(data_points) > 0:
+        return {
+            "n": int(np.size(data_points)),
+            "mean": float(np.mean(data_points)),
+            "median": float(np.median(data_points)),
+            "min": int(np.min(data_points)),
+            "max": int(np.max(data_points)),
+        }
+    else:
+        return {
+            "n": 0,
+            "mean": np.nan,
+            "median": np.nan,
+            "min": np.nan,
+            "max": np.nan,
+        }
+
+
+def compute_single_tracker_stats(
+    single_tracker_markers: Dict[str, Any]
+) -> Dict[str, Stats]:
+    """Computes summary statistics for a single tracker."""
+    tracker_stats = {}
+    for marker in single_tracker_markers["markers"]:
+        tracker_stats[marker["marker"]] = compute_summary_stats(
+            marker["num_preceding_user_turns"]
+        )
+    return tracker_stats
+
+
+def compute_multi_tracker_stats(
+    multi_tracker_markers: list,
+) -> Tuple[Dict, Dict[Any, Dict[str, Stats]]]:
```

Comment:
Do you have the error it's throwing up? I have some guesses why it might be doing it (and some of them are resolved by just not touching anything), but it would be good to know what it's doing to see if we can resolve it